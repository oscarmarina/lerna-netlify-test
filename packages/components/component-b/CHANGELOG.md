# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.0.2](https://github.com/oscarmarina/lerna-netlify-test/compare/@blockquote-playground/component-b@1.0.1...@blockquote-playground/component-b@1.0.2) (2025-10-26)

**Note:** Version bump only for package @blockquote-playground/component-b





## 1.0.1 (2025-10-01)


### Bug Fixes

* **component-b:** update counter ([f60ba57](https://github.com/oscarmarina/lerna-netlify-test/commit/f60ba57b26e60150c3fcf1c68ff456e220fb8a57))



# 0.0.0 (2025-10-01)





# 1.0.0 (2025-09-30)

**Note:** Version bump only for package @blockquote-playground/component-b
