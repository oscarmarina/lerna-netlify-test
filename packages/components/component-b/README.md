![Lit](https://img.shields.io/badge/lit-3.0.0-blue.svg)

## `<component-b>`
An example element.


### `src/ComponentB.js`:

#### class: `ComponentB`, `component-b`

##### Fields

| Name      | Privacy | Type     | Default       | Description                                      | Inherited From |
| --------- | ------- | -------- | ------------- | ------------------------------------------------ | -------------- |
| `heading` | public  | `string` | `'Hey there'` | The heading to say "Hello" to.                   |                |
| `counter` | public  | `number` | `10`          | The number of times the button has been clicked. |                |

##### Methods

| Name       | Privacy | Description        | Parameters        | Return   | Inherited From |
| ---------- | ------- | ------------------ | ----------------- | -------- | -------------- |
| `sayHello` |         | Formats a greeting | `heading: string` | `string` |                |

##### Events

| Name            | Type          | Description                      | Inherited From |
| --------------- | ------------- | -------------------------------- | -------------- |
| `counterchange` | `CustomEvent` | Indicates when the count changes |                |

##### Attributes

| Name      | Field   | Inherited From |
| --------- | ------- | -------------- |
| `heading` | heading |                |
| `counter` | counter |                |

##### Slots

| Name | Description             |
| ---- | ----------------------- |
|      | This element has a slot |

<details><summary>Private API</summary>

##### Methods

| Name       | Privacy | Description | Parameters | Return | Inherited From |
| ---------- | ------- | ----------- | ---------- | ------ | -------------- |
| `#onClick` | private |             |            |        |                |

</details>

<hr/>

#### Exports

| Kind | Name         | Declaration | Module            | Package |
| ---- | ------------ | ----------- | ----------------- | ------- |
| `js` | `ComponentB` | ComponentB  | src/ComponentB.js |         |

### `src/index.js`:

#### Exports

| Kind | Name         | Declaration | Module          | Package |
| ---- | ------------ | ----------- | --------------- | ------- |
| `js` | `ComponentB` | ComponentB  | ./ComponentB.js |         |

### `src/define/component-b.js`:

#### Exports

| Kind                        | Name          | Declaration | Module             | Package |
| --------------------------- | ------------- | ----------- | ------------------ | ------- |
| `custom-element-definition` | `component-b` | ComponentB  | /src/ComponentB.js |         |

### `src/styles/component-b-styles.css.js`:

#### Variables

| Name     | Description | Type |
| -------- | ----------- | ---- |
| `styles` |             |      |

<hr/>

#### Exports

| Kind | Name     | Declaration | Module                               | Package |
| ---- | -------- | ----------- | ------------------------------------ | ------- |
| `js` | `styles` | styles      | src/styles/component-b-styles.css.js |         |
