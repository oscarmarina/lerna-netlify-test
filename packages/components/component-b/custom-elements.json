{
  "schemaVersion": "1.0.0",
  "readme": "",
  "modules": [
    {
      "kind": "javascript-module",
      "path": "src/ComponentB.js",
      "declarations": [
        {
          "kind": "class",
          "description": "![Lit](https://img.shields.io/badge/lit-3.0.0-blue.svg)\n\n## `<component-b>`\nAn example element.",
          "name": "ComponentB",
          "slots": [
            {
              "description": "This element has a slot",
              "name": ""
            }
          ],
          "members": [
            {
              "kind": "method",
              "name": "#onClick",
              "privacy": "private"
            },
            {
              "kind": "method",
              "name": "sayHello",
              "parameters": [
                {
                  "name": "heading",
                  "description": "The heading to say \"Hello\" to",
                  "type": {
                    "text": "string"
                  }
                }
              ],
              "description": "Formats a greeting",
              "return": {
                "type": {
                  "text": "string"
                }
              }
            },
            {
              "kind": "field",
              "name": "heading",
              "privacy": "public",
              "type": {
                "text": "string"
              },
              "description": "The heading to say \"Hello\" to.",
              "default": "'Hey there'",
              "attribute": "heading"
            },
            {
              "kind": "field",
              "name": "counter",
              "privacy": "public",
              "type": {
                "text": "number"
              },
              "description": "The number of times the button has been clicked.",
              "default": "10",
              "attribute": "counter"
            }
          ],
          "events": [
            {
              "name": "counterchange",
              "type": {
                "text": "CustomEvent"
              },
              "description": "Indicates when the count changes"
            }
          ],
          "attributes": [
            {
              "name": "heading",
              "type": {
                "text": "string"
              },
              "description": "The heading to say \"Hello\" to.",
              "default": "'Hey there'",
              "fieldName": "heading"
            },
            {
              "name": "counter",
              "type": {
                "text": "number"
              },
              "description": "The number of times the button has been clicked.",
              "default": "10",
              "fieldName": "counter"
            }
          ],
          "superclass": {
            "name": "LitElement",
            "package": "lit"
          },
          "tagName": "component-b",
          "customElement": true
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "ComponentB",
          "declaration": {
            "name": "ComponentB",
            "module": "src/ComponentB.js"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/index.js",
      "declarations": [],
      "exports": [
        {
          "kind": "js",
          "name": "ComponentB",
          "declaration": {
            "name": "ComponentB",
            "module": "./ComponentB.js"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/define/component-b.js",
      "declarations": [],
      "exports": [
        {
          "kind": "custom-element-definition",
          "name": "component-b",
          "declaration": {
            "name": "ComponentB",
            "module": "/src/ComponentB.js"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/styles/component-b-styles.css.js",
      "declarations": [
        {
          "kind": "variable",
          "name": "styles",
          "default": "css` :host { display: block; box-sizing: border-box; background-color: gainsboro; } :host([hidden]), [hidden] { display: none !important; } *, *::before, *::after { box-sizing: inherit; } `"
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "styles",
          "declaration": {
            "name": "styles",
            "module": "src/styles/component-b-styles.css.js"
          }
        }
      ]
    }
  ]
}
