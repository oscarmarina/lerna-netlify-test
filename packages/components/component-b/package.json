{
  "name": "@blockquote-playground/component-b",
  "version": "1.0.0",
  "description": "Webcomponent component-b following open-wc recommendations",
  "keywords": [
    "lit",
    "web-component",
    "lit-element"
  ],
  "license": "MIT",
  "author": "component-b",
  "type": "module",
  "exports": {
    ".": {
      "default": "./src/index.js"
    },
    "./package.json": {
      "default": "./package.json"
    },
    "./ComponentB.js": {
      "default": "./src/ComponentB.js"
    },
    "./component-b.js": {
      "default": "./src/define/component-b.js"
    },
    "./index.js": {
      "default": "./src/index.js"
    },
    "./component-b-styles.css.js": {
      "default": "./src/styles/component-b-styles.css.js"
    }
  },
  "main": "src/index.js",
  "module": "src/index.js",
  "files": [
    "/src/",
    "!/**/*.scss"
  ],
  "scripts": {
    "analyze": "cem analyze --litelement --globs \"src/**/*.{js,ts}\" \"index.js\"",
    "build": "echo \"This is not a TypeScript project, so no need to build.\"",
    "dev:vite": "vite build",
    "format": "npm run format:eslint && npm run format:prettier && npm run format:stylelint",
    "format:eslint": "eslint . --fix",
    "format:prettier": "prettier \"**/*.{js,ts,css,scss,html,json}\" --write --ignore-path .prettierignore",
    "format:stylelint": "stylelint \"{demo,src}/**/*.{js,ts,css,scss,html}\" --fix --ignore-path .prettierignore",
    "postinstall": "npm run sort:package",
    "prepare": "husky",
    "preview:vite": "vite preview",
    "sass:watch": "sass-style-template",
    "sort:package": "npx sort-package-json",
    "start": "concurrently -k -r \"npm:vite\" \"npm:sass:watch\"",
    "test": "vitest --run --browser.headless",
    "test:watch": "vitest",
    "vite": "vite"
  },
  "lint-staged": {
    "**/*.{js,ts,html}": [
      "npm run format"
    ],
    "**/*.json": [
      "npm run format:prettier"
    ],
    "**/*.{scss,css}": [
      "npm run format:prettier",
      "npm run format:stylelint"
    ]
  },
  "prettier": {
    "bracketSpacing": false,
    "printWidth": 100,
    "singleQuote": true,
    "trailingComma": "es5",
    "overrides": [
      {
        "files": "*.{scss,css}",
        "options": {
          "singleQuote": false
        }
      },
      {
        "files": "*.json",
        "options": {
          "parser": "json-stringify"
        }
      }
    ]
  },
  "stylelint": {
    "extends": "stylelint-config-standard-scss",
    "allowEmptyInput": true,
    "rules": {
      "custom-property-pattern": null,
      "no-duplicate-selectors": null,
      "color-function-notation": null,
      "alpha-value-notation": null,
      "media-feature-name-no-unknown": null,
      "comment-empty-line-before": null,
      "declaration-empty-line-before": null,
      "property-no-vendor-prefix": null,
      "selector-attribute-quotes": null,
      "value-no-vendor-prefix": null
    },
    "overrides": [
      {
        "files": [
          "**/*.html"
        ],
        "customSyntax": "postcss-html"
      },
      {
        "files": [
          "**/*.{ts,js}"
        ],
        "customSyntax": "postcss-lit"
      }
    ]
  },
  "dependencies": {
    "lit": "^3.3.1"
  },
  "devDependencies": {
    "@blockquote/cem-to-markdown-readme": "^1.0.0",
    "@blockquote/rollup-plugin-total-bundlesize": "^2.2.0",
    "@blockquote/sass-style-template": "^5.0.2",
    "@custom-elements-manifest/analyzer": "^0.10.8",
    "@html-eslint/eslint-plugin": "^0.47.0",
    "@html-eslint/parser": "^0.47.0",
    "@open-wc/semantic-dom-diff": "^0.20.1",
    "@open-wc/testing-helpers": "^3.0.1",
    "@types/mocha": "^10.0.10",
    "@typescript-eslint/parser": "^8.45.0",
    "@vitest/browser": "^3.2.4",
    "@vitest/coverage-istanbul": "^3.2.4",
    "@vitest/coverage-v8": "^3.2.4",
    "@vitest/ui": "^3.2.4",
    "chai-a11y-axe": "^1.5.0",
    "concurrently": "^9.2.1",
    "eslint": "^9.36.0",
    "eslint-config-prettier": "^10.1.8",
    "eslint-plugin-html": "^8.1.3",
    "eslint-plugin-import": "^2.32.0",
    "eslint-plugin-lit": "^2.1.1",
    "eslint-plugin-lit-a11y": "^5.1.1",
    "eslint-plugin-wc": "^3.0.1",
    "globals": "^16.4.0",
    "husky": "^9.1.7",
    "lint-staged": "^16.2.3",
    "playwright": "^1.55.1",
    "postcss-html": "^1.8.0",
    "postcss-lit": "^1.3.1",
    "prettier": "^3.6.2",
    "rollup-plugin-copy": "^3.5.0",
    "rollup-plugin-node-externals": "^8.1.1",
    "sort-package-json": "^3.4.0",
    "stylelint": "^16.24.0",
    "stylelint-config-standard-scss": "^16.0.0",
    "tinyglobby": "^0.2.15",
    "typescript-eslint": "^8.45.0",
    "vite": "^7.1.7",
    "vitest": "^3.2.4"
  },
  "publishConfig": {
    "access": "public"
  },
  "customElements": "custom-elements.json"
}
