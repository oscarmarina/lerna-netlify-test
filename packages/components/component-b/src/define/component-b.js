import {ComponentB} from '../ComponentB.js';

window.customElements.define('component-b', ComponentB);
