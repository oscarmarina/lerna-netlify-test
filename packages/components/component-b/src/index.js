export {ComponentB} from './ComponentB.js';
