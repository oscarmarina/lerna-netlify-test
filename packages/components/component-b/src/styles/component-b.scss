:host {
  display: block;
  box-sizing: border-box;
  background-color: gainsboro;
}

:host([hidden]),
[hidden] {
  display: none !important;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}
