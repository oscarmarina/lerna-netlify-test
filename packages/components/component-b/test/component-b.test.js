import {describe, it, expect, beforeAll, afterAll, beforeEach, afterEach, vi, chai} from 'vitest';
import {utils} from 'vitest/browser';
import {fixture, fixtureCleanup} from '@open-wc/testing-helpers';
import {chaiA11yAxe} from 'chai-a11y-axe';
import {getDiffableHTML} from '@open-wc/semantic-dom-diff/get-diffable-html.js';
import {html} from 'lit';
import '../src/define/component-b.js';

chai.use(chaiA11yAxe);

describe('ComponentB', () => {
  /**
   * @type {import('../src/index').ComponentB}
   */
  let el;
  let elShadowRoot;
  let elLocator;

  describe('Semantic Dom and a11y', () => {
    beforeAll(async () => {
      el = await fixture(html`<component-b>light-dom</component-b>`);
      elShadowRoot = el?.shadowRoot?.innerHTML;
      elLocator = utils.getElementLocatorSelectors(el);
    });

    afterAll(() => {
      fixtureCleanup();
    });

    it('has a default heading "Hey there" and counter 15', () => {
      const button = elLocator.getByText('Counter: 15').query();
      const heading = elLocator.getByText('Hey there').query();
      expect(button).toBeTruthy();
      expect(heading).toBeTruthy();
    });

    it('SHADOW DOM - Structure test', () => {
      expect(getDiffableHTML(elShadowRoot)).toMatchSnapshot('SHADOW DOM');
    });

    it('LIGHT DOM - Structure test', () => {
      expect(getDiffableHTML(el, {ignoreAttributes: ['id']})).toMatchSnapshot('LIGHT DOM');
    });

    it('a11y', async () => {
      await expect(el).accessible();
    });
  });

  describe('Events ', () => {
    beforeEach(async () => {
      el = await fixture(html`<component-b></component-b>`);
      elLocator = utils.getElementLocatorSelectors(el);
    });

    afterEach(() => {
      fixtureCleanup();
    });

    it('should increment value on click', async () => {
      const button = elLocator.getByText('Counter: 15');
      const elButton = button.query();
      await button.dblClick();
      await el.updateComplete;
      expect(elButton.textContent).toContain('Counter: 17');
    });

    it('counterchange event is dispatched - vi', async () => {
      const spyEvent = vi.spyOn(el, 'dispatchEvent');
      const button = elLocator.getByText('Counter: 15');
      await button.click();
      expect(spyEvent).toHaveBeenCalledWith(expect.objectContaining({type: 'counterchange'}));
    });
  });

  describe('Override ', () => {
    beforeAll(async () => {
      el = await fixture(html`<component-b heading="attribute heading"></component-b>`);
      elLocator = utils.getElementLocatorSelectors(el);

      return () => {
        fixtureCleanup();
      };
    });

    it('can override the heading via attribute', () => {
      expect(el.heading).toBe('attribute heading');
    });
  });
});
