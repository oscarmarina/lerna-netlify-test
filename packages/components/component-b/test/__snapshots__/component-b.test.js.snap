// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ComponentB > Semantic Dom and a11y > LIGHT DOM - Structure test > LIGHT DOM 1`] = `
"<component-b>
  light-dom
</component-b>
"
`;

exports[`ComponentB > Semantic Dom and a11y > SHADOW DOM - Structure test > SHADOW DOM 1`] = `
"<h1>
  Hello, Hey there!
</h1>
<button>
  Counter: 10
</button>
<hr>
<slot>
</slot>
"
`;
